skrifa/src/meta/metrics.rs

```diff
@@ -0,0 +1,434 @@
+        }
+        let gid_index = glyph_id.to_u16() as usize;
+        let mut lsb = self
+            .h_metrics
+            .get(gid_index)
+            .map(|metric| metric.side_bearing())
+            .unwrap_or_else(|| {
+                self.lsbs
+                    .get(gid_index.saturating_sub(self.h_metrics.len()))
+                    .map(|lsb| lsb.get())
+                    .unwrap_or_default()
+            }) as i32;
+        if let Some(hvar) = &self.hvar {
+            lsb += hvar
+                .lsb_delta(glyph_id, self.coords)
+                // FreeType truncates metric deltas...
+                .map(|delta| delta.to_f64() as i32)
+                .unwrap_or(0);
+        }
+        Some(lsb as f32 * self.scale)
+    }
+
+    /// Returns the bounding box for the specified glyph.
+    ///
+    /// Note that variations are not reflected in the bounding box returned by
+    /// this method.
+    pub fn bounds(&self, glyph_id: GlyphId) -> Option<BoundingBox> {
+        let (loca, glyf) = self.loca_glyf.as_ref()?;
+        Some(match loca.get_glyf(glyph_id, glyf).ok()? {
+            Some(glyph) => BoundingBox {
+                x_min: glyph.x_min() as f32 * self.scale,
+                y_min: glyph.y_min() as f32 * self.scale,
+                x_max: glyph.x_max() as f32 * self.scale,
+                y_max: glyph.y_max() as f32 * self.scale,
+            },
+            // Empty glyphs have an empty bounding box
+            None => BoundingBox::default(),
+        })
+    }
+}
+
+fn compute_scale(size: Option<f32>, upem: f32) -> f32 {
+    let size = size.unwrap_or(0.0).abs();
+    if size == 0.0 || upem == 0.0 {
+        1.0
+    } else {
+        size / upem
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::MetadataProvider as _;
+    use read_fonts::test_data::test_fonts::{SIMPLE_GLYF, VAZIRMATN_VAR};
+    use read_fonts::FontRef;
+
+    #[test]
+    fn metrics() {
+        let font = FontRef::new(SIMPLE_GLYF).unwrap();
+        let metrics = font.metrics(None, &[]);
+        let expected = Metrics {
+            units_per_em: 1024,
+            glyph_count: 3,
+            bounds: Some(BoundingBox {
+                x_min: 51.0,
+                y_min: -250.0,
+                x_max: 998.0,
+                y_max: 950.0,
+            }),
+            average_width: Some(1275.0),
+            max_width: None,
+            x_height: Some(512.0),
+            cap_height: Some(717.0),
+            is_monospace: false,
+            italic_angle: 0.0,
+            ascent: 950.0,
+            descent: -250.0,
+            leading: 0.0,
+            underline: None,
+            strikeout: Some(Decoration {
+                offset: 307.0,
+                thickness: 51.0,
+            }),
+        };
+        assert_eq!(metrics, expected);
+    }
+
+    #[test]
+    fn metrics_missing_os2() {
+        let font = FontRef::new(VAZIRMATN_VAR).unwrap();
+        let metrics = font.metrics(None, &[]);
```

Comment:
Not crazy about this, I feel like I'd have to lookup the metrics fn to know what this means.

Reply:
As above, I’ll try to improve the signature. I think removing the option around size would help.

Reply:
Per f2f, this code now looks like:
```rust
let metrics = font.metrics(Size::unscaled(), NormalizedCoords::default());
```